storage/storage_dest.go

```diff
@@ -469,7 +496,14 @@ func (s *storageImageDestination) computeID(m manifest.Manifest) string {
 		// the diffID list.
 	case *manifest.OCI1:
 		for _, l := range m.Layers {
-			diffIDs = append(diffIDs, l.Digest)
+			if l.Digest != "" {
+				diffIDs = append(diffIDs, l.Digest)
+			} else {
+				tocDigest, err := chunkedToc.GetTOCDigest(l.Annotations)
+				if err == nil && tocDigest != nil {
+					diffIDs = append(diffIDs, *tocDigest)
+				}
+			}
```

Comment:
This the critical part:

No, AFAICS this must be dependent on _how the layer was actually pulled_, not just based on what the manifest says. The same manifest with N chunked layers can be pulled 2^N different ways, depending on how the registry decides to accept/reject ranged requests in each case.

Reply:
changed to only use the TOC id if it is the one we used to refer to the layer